packages/ember-metal/lib/expand_properties.js

```diff
@@ -37,26 +36,26 @@ var END_WITH_EACH_REGEX = /\.@each$/;
   expansion, and is passed the expansion.
 */
 export default function expandProperties(pattern, callback) {
+  if (typeof pattern !== 'string') {
+    throw new EmberError('A computed property key must be a string');
```

Comment:
Can we change this to be an `assert` instead?  I'd hate to continue doing this `indexOf` in prod builds just to provide a more helpful message than the error that would already be thrown...